crypto/rand/rand_lib.c

```diff
@@ -638,6 +650,10 @@ int rand_pool_add(RAND_POOL *pool,
     }
 
     if (len > 0) {
+        if (pool->alloc_len > pool->len && pool->buffer + pool->len == buffer) {
```

Comment:
Most of the other checks (including those already present) are self-explanatory. This one however, is a bit mysterious and deserves an explaining comment.

Reply:
_(essentially what you said above in this pr thread)_